src/app/data-planes/components/RuleEntryList.vue

```diff
@@ -4,116 +4,114 @@
     multiple-open
   >
     <template
-      v-for="policies in [props.rules.reduce<Record<string, Rule[]>>((prev, item) => {
-        if(typeof prev[item.type] === 'undefined') {
-          prev[item.type] = []
-        }
-        prev[item.type].push(item)
-        return prev
-      }, {})]"
+      v-for="policies in [Object.groupBy(props.rules, (item, i) => item.type)]"
       :key="policies"
     >
-      <AccordionItem
+      <template
         v-for="(items, type) in policies"
         :key="type"
       >
-        <template #accordion-header>
-          <h3 class="policy-type-heading">
-            <PolicyTypeTag :policy-type="type">
-              {{ type }}
-            </PolicyTypeTag>
-          </h3>
-        </template>
-
-        <template #accordion-content>
-          <template
-            v-for="hasMatchers in [items.some((item) => item.matchers.length > 0)]"
-            :key="hasMatchers"
-          >
-            <div class="policy-list">
-              <AppCollection
-                class="policy-type-table"
-                :class="{
-                  'has-matchers': hasMatchers,
-                }"
-                :total="items.length"
-                :items="items"
-                :headers="[
-                  ...(hasMatchers ? [{ label: 'Matchers', key: 'matchers' }] : []),
-                  { label: 'Origin policies', key: 'origins' },
-                  { label: 'Conf', key: 'config' },
-                ]"
-              >
-                <template
-                  #matchers="{ row }"
+        <AccordionItem
+          v-if="items"
```

Comment:
**Question**: How come this is change was needed/made? Shouldn’t this change be a simple replacement of "groupBy" code that has no other impact?

Reply:
I wrapped the AccordianThing in a further `template` because I unfortunately also now need to do a v-if="" otherwise TS complains, I think this is the same issue as our conversation above just approached differently.

Reply:
See also https://github.com/kumahq/kuma-gui/pull/2287#discussion_r1524855069